### ISSUE TITLE
[Docs]: Recommended directory structure of root directory with explination showing torrent lifecycle.

### ISSUE THREAD

**Issue by Flute7** [StuffAnThings/qbit_manage#318]
### What is the improvement or update you wish to see?

The documentation should include a comprehensive sample directory structure of the root folder. Additionally, a description of the typical movement of a torrent file within this structure; over its lifecycle would assist with understanding and setup.

### Is there any context that might help us understand?

The following questions are not easily answered using the documentation.

- It is unclear how/where the program checks for hard-links (does it just check the files hard-link status or does it verify by checking a specific location for a matching hard-link?).
- How are torrents moved between the incomplete and complete categories (i.e. 'movies' and 'movies-completed'). Do these categories point to different directories? qBittorrent doesn't have the functionality to move completed torrents from 'movies' to 'movies-completed' so what program is performing this move?
- Is it acceptable to have no 'completed' directory and just perform the no-hardlinks processing on the original category (i.e. 'movies' instead of 'movies-completed')?

My apologies if this is a question dump. I wanted to resolve my lack of understanding in a constructive way that might help to improve the documentation.

**Comment by bakerboy448** [comment]
> comprehensive sample directory structure of the root folder

impossible...everyone's setup is unique.....
`  # root_dir var: </your/path/here/>  # Root downloads directory used to check for orphaned files, noHL, and RecycleBin.`
it's clearly explained and defined as the base folder for all your downloads.
`  root_dir: "/data/torrents/"`  the example follows the recommended hardlink structure

> How are torrents moved between the incomplete and complete categories (i.e. 'movies' and 'movies-completed'). Do these categories point to different directories? qBittorrent doesn't have the functionality to move completed torrents from 'movies' to 'movies-completed' so what program is performing this move?

Qbit manage has nothing to do with incomplete folders.  there is no such thing as incomplete categories.
If you're talking Starr post-import categories those have nothing to do with qbm nor does anything anywhere indicate as such.  Starr apps post-import cats are the cat Starr moves the torrent to after import. Exactly as Starr docs and UI state.

> Is it acceptable to have no 'completed' directory and just perform the no-hardlinks processing on the original category (i.e. 'movies' instead of 'movies-completed')?

This makes no sense and has nothing to do with QBM.  Incomplete torrents have no relevance to no hardlinks.

@bobokun can likely close; I see nothing to act on here and there's no need to document 3rd party apps that have nothing to do with qbm

**Comment by bobokun** [comment]
You can view more information on https://trash-guides.info/Hardlinks/How-to-setup-for/

**Comment by Flute7** [comment]
> > comprehensive sample directory structure of the root folder
> 
> impossible...everyone's setup is unique..... ` # root_dir var: </your/path/here/> # Root downloads directory used to check for orphaned files, noHL, and RecycleBin.` it's clearly explained and defined as the base folder for all your downloads. ` root_dir: "/data/torrents/"` the example follows the recommended hardlink structure
> 
> > How are torrents moved between the incomplete and complete categories (i.e. 'movies' and 'movies-completed'). Do these categories point to different directories? qBittorrent doesn't have the functionality to move completed torrents from 'movies' to 'movies-completed' so what program is performing this move?
> 
> Qbit manage has nothing to do with incomplete folders. there is no such thing as incomplete categories. If you're talking Starr post-import categories those have nothing to do with qbm nor does anything anywhere indicate as such. Starr apps post-import cats are the cat Starr moves the torrent to after import. Exactly as Starr docs and UI state.
> 
> > Is it acceptable to have no 'completed' directory and just perform the no-hardlinks processing on the original category (i.e. 'movies' instead of 'movies-completed')?
> 
> This makes no sense and has nothing to do with QBM. Incomplete torrents have no relevance to no hardlinks.
> 
> @bobokun can likely close; I see nothing to act on here and there's no need to document 3rd party apps that have nothing to do with qbm

Perhaps my explanation was just incredibly bad but every question I had seems to be misinterpreted. Let me try again to narrow down the specifics of what the documentation does not make clear.

The Trash-Guides directory structure does not include complete and incomplete directories. However, my confusion lies in the fact that the sample config references categories named 'movies' and 'movies-completed'. 
Are these categories linked to the same directory? If they are, what is changing the torrents category on completion?
Are these categories linked to the different directories? If they are, what is changing the torrents category and directory on completion?

**Comment by bakerboy448** [comment]
Torrent clients have no concept of incomplete categories. No idea why you believe they are.  Names of categories are arbitrary.

It also appears you didn't read my response  at all as it already answered questions.

> there is no such thing as incomplete categories.

> If you're talking Starr post-import categories those have nothing to do with qbm nor does anything anywhere indicate as such. Starr apps post-import cats are the cat Starr moves the torrent to after import. Exactly as Starr docs and UI state.

**Comment by Flute7** [comment]
> Torrent clients have no concept of incomplete categories. No idea why you believe they are. Names of categories are arbitrary....whatever categories you have configured in qbit - you must configure in QBm. QBM has nothing to do at all in anyway with what categories you setup in qbt or what pre/post import categories you configure in the Starr apps.
> 
> No where does anything say an incomplete category is used or exists nor does that concept except within any torrent client. You're confusing pre/post import categories with Usenet incomplete and complete folders.
> 
> It also appears you didn't read my response at all as it already answered questions.
> 
> > there is no such thing as incomplete categories.
> 
> > If you're talking Starr post-import categories those have nothing to do with qbm nor does anything anywhere indicate as such. Starr apps post-import cats are the cat Starr moves the torrent to after import. Exactly as Starr docs and UI state.

I don't use the Starr apps or Usenet. I'm not confusing anything related to them.
Are you saying that the documentation uses two different names, 'movies' and 'movies-completed' to reference the same thing? If this is the case can the documentation please be modified for consistency.

**Comment by bakerboy448** [comment]
The documentation is just examples...you adjust them to fit your specific setup...including configuring only the categories you have in the config....most folks using QBM use radarr/sonarr/etc which many use pre-import and post-import (after start has imported it to the library) cat

**Comment by Flute7** [comment]
> The documentation is just examples...you adjust them to fit your specific setup...including configuring only the categories you have in the config....most folks using QBM use radarr/sonarr/etc which many use pre-import and post-import (after start has imported it to the library) cat hence movies and movies-complete.
> 
> Not sure why you're so hung up on two categories with arbitrary names.

Thank you, I really appreciate the explanation. I was hung up on the names because they implied functionality that I thought I didn't understand.

**Comment by putnam** [comment]
What a hostile and attitude-filled response to what was an honest and friendly question.
